### PR TITLE
:sparkles: add `StrictMerge` option and improve list-limit handling in decoder tests

### PR DESCRIPTION
**Pull request by techouse**
This pull request primarily updates the `qs` npm package dependency to version 6.15.2 and significantly expands and refines the test suite for the `QsNet` decoding logic. The test changes focus on stricter and more accurate handling of array/list limits, merging strategies, and edge cases such as comma-separated values and bracket notation. These improvements help ensure the decoder's behavior closely matches the intended and documented behavior, especially in complex or ambiguous scenarios.

**Dependency Update:**
- Upgraded the `qs` npm package from version 6.15.1 to 6.15.2 in both `package.json` and `pnpm-lock.yaml`, ensuring the latest bug fixes and features from the upstream library are included. [[1]](diffhunk://#diff-33714f3b51fe0edf079ff43ee28ff5950d3a22c45469e74c00a1a5fc99698903L9-R9) [[2]](diffhunk://#diff-d7474e380e2258d8c6d903c74bd0919cd41b5c8820608d3d8e2ec8f14a2ff6d3L12-R13) [[3]](diffhunk://#diff-d7474e380e2258d8c6d903c74bd0919cd41b5c8820608d3d8e2ec8f14a2ff6d3L72-R73) [[4]](diffhunk://#diff-d7474e380e2258d8c6d903c74bd0919cd41b5c8820608d3d8e2ec8f14a2ff6d3L150-R150)

**Decode Options and Merge Behavior:**
- Added and tested the `StrictMerge` option in `DecodeOptions`, with new tests verifying both the default and legacy (false) behaviors for merging object and primitive values under the same key. [[1]](diffhunk://#diff-7faa428ecf2b0836aa88101a16f5d62b0b2f122e2b45e843b14435c946424ad4R32) [[2]](diffhunk://#diff-7faa428ecf2b0836aa88101a16f5d62b0b2f122e2b45e843b14435c946424ad4R53) [[3]](diffhunk://#diff-7faa428ecf2b0836aa88101a16f5d62b0b2f122e2b45e843b14435c946424ad4R96) [[4]](diffhunk://#diff-7faa428ecf2b0836aa88101a16f5d62b0b2f122e2b45e843b14435c946424ad4R114) [[5]](diffhunk://#diff-7aebf87dd67b9e97f3a629a859b0aaa0d42c4ba4ed12f41991c943b62241fbcdR2667-R2742) [[6]](diffhunk://#diff-7aebf87dd67b9e97f3a629a859b0aaa0d42c4ba4ed12f41991c943b62241fbcdL4405-R4502)
- Improved handling and tests for duplicate keys, ensuring bracket notation always combines values appropriately regardless of the `Duplicates` setting.

**Array/List Limit Handling:**
- Refined tests to ensure that explicit and implicit array indices respect the `ListLimit` option, including correct conversion to maps when limits are exceeded and proper exception throwing when `ThrowOnLimitExceeded` is set. [[1]](diffhunk://#diff-7aebf87dd67b9e97f3a629a859b0aaa0d42c4ba4ed12f41991c943b62241fbcdL1185-R1207) [[2]](diffhunk://#diff-7aebf87dd67b9e97f3a629a859b0aaa0d42c4ba4ed12f41991c943b62241fbcdL1217-R1221) [[3]](diffhunk://#diff-7aebf87dd67b9e97f3a629a859b0aaa0d42c4ba4ed12f41991c943b62241fbcdL1835-R1844) [[4]](diffhunk://#diff-7aebf87dd67b9e97f3a629a859b0aaa0d42c4ba4ed12f41991c943b62241fbcdR2958-R2969) [[5]](diffhunk://#diff-7aebf87dd67b9e97f3a629a859b0aaa0d42c4ba4ed12f41991c943b62241fbcdL3436-R3555)

**Comma-Separated Value Handling:**
- Overhauled tests for comma-separated values, ensuring that when the sum of values exceeds the list limit, the decoder now converts the result to a map (overflow map) instead of truncating, and all values are preserved. This behavior is now consistently tested across different scenarios, including bracketed keys and nested lists. [[1]](diffhunk://#diff-7aebf87dd67b9e97f3a629a859b0aaa0d42c4ba4ed12f41991c943b62241fbcdL4650-R4772) [[2]](diffhunk://#diff-7aebf87dd67b9e97f3a629a859b0aaa0d42c4ba4ed12f41991c943b62241fbcdL4664-R4798) [[3]](diffhunk://#diff-7aebf87dd67b9e97f3a629a859b0aaa0d42c4ba4ed12f41991c943b62241fbcdL4683-R4824) [[4]](diffhunk://#diff-7aebf87dd67b9e97f3a629a859b0aaa0d42c4ba4ed12f41991c943b62241fbcdL4702-R4842) [[5]](diffhunk://#diff-7aebf87dd67b9e97f3a629a859b0aaa0d42c4ba4ed12f41991c943b62241fbcdR4865-R4880) [[6]](diffhunk://#diff-7aebf87dd67b9e97f3a629a859b0aaa0d42c4ba4ed12f41991c943b62241fbcdL5391-R5566) [[7]](diffhunk://#diff-7aebf87dd67b9e97f3a629a859b0aaa0d42c4ba4ed12f41991c943b62241fbcdL5401-R5588) [[8]](diffhunk://#diff-7aebf87dd67b9e97f3a629a859b0aaa0d42c4ba4ed12f41991c943b62241fbcdL5417-R5603)

**Edge Case and Encoding Tests:**
- Added new tests to ensure correct decoding of percent-encoded bracket characters and that literal percent sequences are not incorrectly decoded, improving robustness for unusual but valid inputs.

These changes enhance the reliability and predictability of the decoder, especially in edge cases and under strict option settings.

## Summary by CodeRabbit

* **New Features**
  * Added StrictMerge option (defaults to wrapping conflicts); bracket-array keys always combine; refined ListLimit semantics; encoder uses configured delimiter when charset sentinel present.

* **Bug Fixes**
  * Preserve percent-encoded bracket text when decoding; improved comma-list null handling and overflow preservation; stricter list-index/overflow behavior.

* **Tests**
  * Expanded coverage for list limits, comma-split overflow, duplicate/merge behaviors, encoding round-trips.

* **Documentation**
  * Clarified StrictMerge, ListLimit, and bracket-array decoding behavior.

* **Chores**
  * Bumped qs dependency to ^6.15.2.

[Review Change Stack](https://app.coderabbit.ai/change-stack/techouse/qs-net/pull/70?utm_source=github_walkthrough&utm_medium=github&utm_campaign=change_stack)

### CODE REVIEW

File: QsNet/Internal/Decoder.cs [hunk 1]
```diff
@@ -45,55 +45,52 @@ private static bool IsLatin1(Encoding e) =>
...
                 }
 
+                if (
+                    enforceListLimit
+                    && list.Count > options.ListLimit
+                )
+                    return Utils.MarkListOverflow(list);
+
                 return list;
             }
```
[the remark this answers is not in the document]
Comment (techouse) [reply QsNet/Internal/Decoder.cs:95]: Acknowledged, but this is intentional qs.js 6.15.2 parity. Users decoding untrusted input should enable `ThrowOnLimitExceeded` and enforce request/body size limits. A separate comma segment cap could be considered as a future non-parity hardening option.

File: QsNet/Internal/Utils.cs [hunk 2]
```diff
@@ -1233,6 +1251,28 @@ internal static object CombineWithLimit(object? a, object? b, DecodeOptions opti
...
+                    numericEntries[idx] = entry.Value;
+                else
+                    target[entry.Key] = entry.Value;
+            }
+
+            foreach (var entry in numericEntries)
+                target[(++nextIndex).ToString(CultureInfo.InvariantCulture)] = entry.Value;
+
+            return MarkOverflow(target, Math.Max(nextIndex, GetOverflowMaxIndex(source)));
+        }
```
[the remark this answers is not in the document]
Comment (techouse) [reply QsNet/Internal/Utils.cs]: This finding is not valid for qs.js parity.
```js
qs.parse('a=1&a=2,3,4,5', { comma: true, arrayLimit: 3 })
// => { a: ["1", { "0": "2", "1": "3", "2": "4", "3": "5" }] }
```
Copilot’s suggested expected shape:
```
{ "0": "1", "1": "2", ... }
```
is not what qs.js returns when the incoming comma value has already overflowed before combine.

Current QsNet behavior matches that branch: `IsOverflow(b) && a is IEnumerable<object?>` only flattens when the existing value is already list-like. With an existing scalar, it falls through to `Combine<object?>`, treating the overflow dictionary as a single appended element, like qs.js `[].concat(a, b)`.

File: QsNet/Internal/Decoder.cs [hunk 3]
```diff
@@ -213,6 +210,7 @@ int currentListLength
                 break;
             }
 
+            var isBracketArrayKey = pos != -1 && rawKey.EndsWith("[]", StringComparison.Ordinal);
```
[the remark this answers is not in the document]
Comment (techouse) [reply QsNet/Internal/Decoder.cs:213]: Verified against qs.js 6.15.2: bare `a[]` does not force-combine like `a[]=` does. It should still respect `Duplicates.First` / `Last` in the bare-key duplicate path, so the current `pos != -1 && rawKey.EndsWith("[]")` check is intentional.
